pennylane/interfaces/torch.py

```diff
@@ -184,7 +184,21 @@ def backward(ctx_, ddy):  # pragma: no cover
     def backward(ctx, dy):  # pragma: no cover
         """Implements the backwards pass QNode vector-Jacobian product"""
         ctx.dy = dy
-        vjp = dy.view(1, -1) @ ctx.jacobian.apply(ctx, *ctx.saved_tensors)
+
+        dyv = dy.view(1, -1)
+        jac_res = ctx.jacobian.apply(ctx, *ctx.saved_tensors)
+
+        # When using CUDA, dyv seems to remain on the GPU, while the result
+        # of jac_res is returned on CPU, even though the saved_tensors arguments are
+        # themselves on the GPU. Check whether this has happened, and move things
+        # back to the GPU if required.
+        if dyv.is_cuda or jac_res.is_cuda:
+            if not dyv.is_cuda:
+                dyv = dyv.cuda()
+            if not jac_res.is_cuda:
+                jac_res = jac_res.cuda()
+
+        vjp = dyv @ jac_res

```

Comment:
🙌 thanks for fixing this @glassnotes!

Quick question, will this need to be fixed up on line 141 as well?

Reply:
Oh that's a really good question. When I ran the example, this was the section that actually threw the error. I've tested on both the transfer learning demo, and the small example in the docs #1225 and both work with the fix. Do you know of a code example that would lead to the earlier part being called?

Reply:
The other backward method is called for Hessian computations. That is, if you have

```python
hess = torch.autograd.functional.hessian(cost_fn, params)
```